client/containers/HomePage.js

```diff
@@ -1,52 +1,71 @@
-import React, { Component } from 'react';
+import React from 'react';
 import PropTypes from 'prop-types';
 import { connect } from 'react-redux';
 import { bindActionCreators } from 'redux';
 import Button from 'components/Button/Button';
-import placeActions from 'actions/placeActions';
-import conditionActions from 'actions/conditionActions';
+import * as placeActions from 'actions/placeActions';
+import * as conditionActions from 'actions/conditionActions';
 import Place from 'components/Place/Place';
 import Condition from 'components/Condition/Condition';
+import { hasCompleteConditions } from '../reducers/conditionReducer';
 
-class HomePage extends Component {
-  handleOnClick = () => {
-    this.props.fetchPlaces(this.props.condition);
-  }
-
-  handleOnConditionChange = (value) => {
-    this.props.setRadius(value);
-  }
-  render() {
-    const { condition, place } = this.props;
-    return (
-      <div className="homePageWrapper">
-        <Place place={place} />
-        <div className="searchWrapper">
-          <Condition condition={condition} action={this.handleOnConditionChange}/>
-          <Button onClick={this.handleOnClick} theme="homepageClick" />
-        </div>
+const HomePage = ({ actions, condition, place }) => {
+  const fetchingCondition = condition.fetching
+    ? 'Getting your Location...'
+    : null;
+  const fetchingPlace = place.fetching ? 'Getting a Place...' : null;
+  const hasError =
+    condition.error || place.error
+      ? 'Encountered an error. Please try again.'
+      : null;
+  return (
+    <div className="homePageWrapper">
+      <div>
+        {fetchingCondition ||
```

Comment:
Im not sure if i understand perfectly your comment how I'm mixing two times since technically I think the || doesn't really evaluate to a boolean in this situation. Moreover, I think the Places.js also had a similar pattern using the &&

Reply:
the thing bothers me is you are trying to render error messages and use the existence of the error messages as a bool to evaluate wether you will render place or not. if you want to keep this pattern something like shouldDisplayPlace ? <Place> : message makes more sense?